config_parser.h

```diff
@@ -4,55 +4,61 @@
 #include <memory>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 class NginxConfig;
 
 // The parsed representation of a single config statement.
 class NginxConfigStatement {
  public:
-  std::string ToString(int depth);
-  std::vector<std::string> tokens_;
-  std::unique_ptr<NginxConfig> child_block_;
+	std::string ToString(int depth);
+	std::vector<std::string> tokens_;
+	std::shared_ptr<NginxConfig> child_block_;
 };
 
 // The parsed representation of the entire config.
 class NginxConfig {
  public:
-  std::string ToString(int depth = 0);
-  std::vector<std::shared_ptr<NginxConfigStatement>> statements_;
+	std::string ToString(int depth = 0);
+	std::vector<std::shared_ptr<NginxConfigStatement>> statements_;
+	
+	// performs a full scan, inefficient, change later for more config options
```

Comment:
We likely only need to use the port once upon startup, so this shouldn't be too much of an issue. For now, I say it's ok to just leave it as is unless issues arise in the future.